broker-daemon/relayer/identity.js

```diff
@@ -77,30 +60,33 @@ class Identity {
    */
 
   /**
-   * Authorize a request to act on a given object by signing its id
+   * Authorize a gRPC Request by adding metadata to it.
    *
    * Specifically, the signature is of the following payload:
+   *  - a string name of the request [ what format? ]
    *  - timestamp of the request, in seconds. A timestamp within +/- 60 seconds of the Relayer's time should be accepted.
    *  - A random nonce of 32 bytes, represented in base64. This nonce should not be repeated, but the Relayer may not reject duplicate nonces older than 24 hours.
-   *  - Id of the object to authorize access to
+   *  - Contents of the request, JSON stringified
    *
    * This payload is joined by commas (','), and signed using the public key of the broker.
    *
    * The request can then be validated by the Relayer as being genuine from the owner of the public key.
-   * @param  {String} id - id to authorize a request for
-   * @return {Authorization}
+   * @param  {string} action action to authorize the request for
+   * @param  {object} params Parameters of the request to authorize
+   * @return {grpc.Metadata} Metadata object with necessary items to verify it
    */
-  authorize (id) {
+  authorize (action, params) {
```

Comment:
Why is the action necessary? Aren't we already going to be hitting the endpoint for the action? It just feels messy to have the code littered with action strings with no good place to centralize them, especially because the actions seem redundant (unless I'm missing something)

Reply:
@treygriffith I think this brings up a fair point. I was actually debating the same thing on my PR in the relayer (although for the relayer, it is a little bit more sane).

We wanted to use `action` as a deterrent of replaying the request, however the signature itself (with timestamp nonce etc) should be enough data that the `action` isn't required.

On the relayer side, we'll need to worry about the action, but we already have this information when the request comes so theres no real benefit to having it on the broker if we dont need it